tide/core/LoggingUtility.h

```diff
@@ -0,0 +1,94 @@
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#ifndef LOGGINGUTILITY_H
+#define LOGGINGUTILITY_H
+
+#include <QObject>
+
+#include "types.h"
+
+class LoggingUtility : public QObject
+{
+    Q_OBJECT
+
+public:
+
+    /** @return the value of interaction counter and resets it. */
+    int getRecentInteractionCounter();
+
+    /** @return the number of currently open windows. */
+    size_t getWindowCount() const;
+
+    /** @return the number of open windows from the start. */
+    size_t getAccumulatedWindowCount() const;
+
+    /** @return the timestamp of opening/closing a window */
+    const QString& getCounterModificationTime() const;
+
+    /** @return the timestamp of last interaction */
+    const QString& getLastInteractionTime() const;
+
+    /** @return the name of last interaction */
+    const QString& getLastInteraction() const;
+
+public slots:
+
+    /** Log the event, update the counters and update the timestamp of last interaction */
+    void contentWindowRemoved();
+
+    /** Log the event, update the counters and update the timestamp of last interaction */
+    void contentWindowAdded( ContentWindowPtr contentWindow );
+
+    /** Log the event, update the counters and update the timestamp of last interaction */
+    void contentWindowMovedToFront();
+
+private:
+    size_t _windowCounter;
```

Comment:
Where are the counters initialized to 0? I don't understand how this worked so far. There must be either a constructor to initialize the values, or use the new C++11 assignment of default values to the members.